eclipse-cdt-cloud/cdt-cloud-blueprint
# Fix OOM on Eclipse CI / Fix Github Workflows

#### What it does
* set NODE_OPTIONS globally
* add webpack config to prevent Terser from running in parallel
  * each spawned worker would have the whole memory available again, which will lead to the build not running oom
* use npm registry instead of yarn registry, since this seems to work more reliable on the windows build node
* disable lint header check for files without MIT license
  * header check tries to enforce MIT license
* Fix path to windows AUT
* Fix path to mac main script

Contributed on behalf of STMicroelectronics

#### How to test
Build should work in low memory environments, like Eclipse CI (see https://ci.eclipse.org/theia/job/TheiaCDTCloudTest/job/jf%252Fmemory/9/)

Github Workflows on this PR should work
* downloading plugins may sometimes fail leading to test errors
